lib/IRGen/IRGenSIL.cpp

```diff
@@ -1987,26 +1987,31 @@ void IRGenSILFunction::visitGraphOperationInst(GraphOperationInst *i) {
          "could not find TensorArrayProtocol protocol");
   assert(outputTensorGroupProto && "could not find TensorGroup protocol");
 
-  if (!llvm::TFDynamicCompilation) {
-    // If we are not in dynamic compilation mode, then deabstraction may not
-    // have transformed the graph_ops into a form that we can lower, so do not
-    // try to lower the graph_ops.
-    // TODO(marcrasi): Once we make deabstraction unnecessary for IRGen, we
-    // should be able to lower everything no matter what the mode is, so remove
-    // this check.
-    const std::string errMessage = "!!! Compiler bug -- graph_op " +
-                              opInfo.getOperationName().str() +
-                              " cannot be lowered to LLVM IR !!!\n";
-    abortOnGraphOp(*this, errMessage.c_str());
-
-    // Finally, we set up our explosion results full of undef values.
+  // Sets the results of this op to undef.
```

Comment:
consider extending this comment on WHY this is useful (not just describing WHAT it does)

Reply:
moved this function into `lowerOpToError` (because of your next comment) and added some explanation of why we're setting the results to undef.